src/Fields/interpolate.jl

```diff
@@ -104,14 +107,14 @@ end
 @inline ϕ₈(ξ, η, ζ) =      ξ  *      η  *      ζ
 
 @inline _interpolate(field, ξ, η, ζ, i, j, k) =
-    @inbounds (  ϕ₁(ξ, η, ζ) * field[i,   j,   k  ]
-               + ϕ₂(ξ, η, ζ) * field[i,   j,   k+1]
-               + ϕ₃(ξ, η, ζ) * field[i,   j+1, k  ]
-               + ϕ₄(ξ, η, ζ) * field[i,   j+1, k+1]
-               + ϕ₅(ξ, η, ζ) * field[i+1, j,   k  ]
-               + ϕ₆(ξ, η, ζ) * field[i+1, j,   k+1]
-               + ϕ₇(ξ, η, ζ) * field[i+1, j+1, k  ]
-               + ϕ₈(ξ, η, ζ) * field[i+1, j+1, k+1])
+    CUDA.@allowscalar @inbounds (  ϕ₁(ξ, η, ζ) * field[i,   j,   k  ]
```

Comment:
maybe it's better to add this to the tests instead of the code itself? Or maybe we can write a function that launches a kernel that interpolates a single value and returns it to be used in the tests? I am a little worried that if we put allowscalar in the code and we forget it we might have performance issues in the future

Reply:
I agree but I thought that this is what you were suggesting. That’s why I sprinkled all these warnings.

Reply:
ah, sorry, I was referring to the tests.

Reply:
done in [f1253dd](https://github.com/CliMA/Oceananigans.jl/pull/2701/commits/f1253dda2cb8c6b0f94b63356fa93aea818b1692)